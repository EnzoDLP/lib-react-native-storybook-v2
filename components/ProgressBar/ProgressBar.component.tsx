import React, { useEffect, useRef } from 'react';
import { View, Animated } from 'react-native';

interface ProgressBarProps {
  progress: number;
  height?: number;
  backgroundColor?: string;
  fillColor?: string;
  duration?: number;
}

const ProgressBar: React.FC<ProgressBarProps> = ({
  progress,
  height = 4,
  backgroundColor = '#0000001a',
  fillColor = '#008859',
  duration = 300,
}) => {
  const animatedWidth = useRef(
    new Animated.Value((progress / 100) * 100)
  ).current;

  useEffect(() => {
    Animated.timing(animatedWidth, {
      toValue: (progress / 100) * 100,
      duration: duration,
      useNativeDriver: false,
    }).start();
  }, [progress]);

  return (
    <View
      style={{
        width: '100%',
        borderRadius: 5,
        overflow: 'hidden',
        height,
        backgroundColor,
      }}
    >
      <Animated.View
        style={{
          height: '100%',
          borderRadius: 100,
          width: animatedWidth.interpolate({
            inputRange: [0, 100],
            outputRange: ['0%', '100%'],
          }),
          backgroundColor: fillColor,
        }}
      />
    </View>
  );
};

export default ProgressBar;
