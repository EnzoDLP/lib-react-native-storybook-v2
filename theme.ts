export const Colors = {
  light: {
    text: {
      primary: '#0F0F0F',
      secondary: '#000000',
      disabled: '#686767',
      onBackground: '#FFFFFF',
      link: '#008859',
      linkHover: '#22543A',
      error: '#D21E1E',
    },
    buttons: {
      primary: '#008859',
      hover: '#22543A',
      disabled: '#E7E7E7',
      onBackground: '#FFFFFF',
    },
    border: {
      primary: '#008859',
    },
    background: {
      light: '#F5F5F5',
      dark: '#686767',
      green: '#E6F3EE',
      default: '#FFFFFF',
      grey: '#e8e8e8',
    },
    input: {
      primary: {
        background: '#FFFFFF',
        border: '#686767',
      },
      focus: {
        background: '#FFFFFF',
        border: '#078213',
      },
      error: {
        background: '#FFE9ED',
        border: '#D21E1E',
      },
      disabled: {
        background: '#E7E7E7',
        border: '#B6B6B6',
      },
    },
  },
};
