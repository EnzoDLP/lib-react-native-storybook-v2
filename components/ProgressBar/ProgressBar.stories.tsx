import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { View } from 'react-native';
import ProgressBar from './ProgressBar.component';
import { Colors } from '../../theme';

const meta = {
  title: 'Components/ProgressBar',
  component: ProgressBar,
  decorators: [
    (Story) => (
      <View style={{ padding: 16 }}>
        <Story />
      </View>
    ),
  ],
} satisfies Meta<typeof ProgressBar>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    progress: 10,
    backgroundColor: Colors.light.background.grey,
    duration: 300,
    fillColor: Colors.light.buttons.primary,
    height: 5,
  },
};
