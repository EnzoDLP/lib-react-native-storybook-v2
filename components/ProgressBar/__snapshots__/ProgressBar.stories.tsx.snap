// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/ProgressBar Default smoke-test 1`] = `
<div class="css-view-175oi2r"
     style="padding: 16px;"
>
  <div aria-label="Progression : 10%"
       role="progressbar"
       aria-valuemax="100"
       aria-valuemin="0"
       aria-valuenow="10"
       class="css-view-175oi2r"
       style="width: 100%; border-radius: 5px; overflow: hidden; height: 5px; background-color: rgb(232, 232, 232);"
  >
    <div class="css-view-175oi2r"
         style="height: 100%; border-radius: 100px; background-color: rgb(0, 136, 89); width: 10%;"
    >
    </div>
  </div>
</div>
`;
